base/containers/checked_iterators.h

```diff
@@ -31,7 +31,10 @@ class CheckedContiguousIterator {
 
   // Required for certain libc++ algorithm optimizations that are not available
   // for NaCl.
-#if defined(_LIBCPP_VERSION) && !BUILDFLAG(IS_NACL)
+// TODO: Remove this when we upgrade to c++20.
+#if (defined(_LIBCPP_VERSION) && !BUILDFLAG(IS_NACL)) || defined(STARBOARD)
```

Comment:
@kaidokert is _LIBCPP_VERSION only defined when `-stdlib=libc++` is explicitly specified? Would we generally want it to be defined for us when compiling above starboard?

Reply:
It's not a predefined macro that would be injected by compiler. Check with
```
clang -stdlib=libc++ -x c++ -dM -E - < /dev/null | egrep -i cpp
```

or 

```
cpp -x c++ -dM < /dev/null | egrep -i cpp 
```

It looks to come from libcpp headers once any of the headers get included, see the note in https://reviews.llvm.org/D134187

So we can expect it to be defined in `stdlib=libc++` builds in files where any of the standard headers are defined. All of Base code seems to rely on this though, and so i think we should ensure ( e.g. even assert somewhere ) that  it's always on for all Cobalt toolchain builds.

Reply:
To doublecheck:

```
echo "#include <cstdint>" | clang -stdlib=libc++ -x c++ -dM -E - | egrep -i cpp_ver
```

outputs: 
```
#define _LIBCPP_VERSION 160006
```

Reply:
Right now `-stdlib=libc++` is only defined for a couple platforms as well as when building skia, but that's it. First we'll have to test that we can actually add it more globally for the cobalt toolchain.

@sherryzy for now could you change this to a `USE_HACKY_COBALT_CHANGES` and make a bug in the taskflow for this TODO (and put it in the code)?

Reply:
Hmm, I don't have much context here. Are there any action items I need to take in this PR?